request.go

```diff
@@ -1106,3 +1106,60 @@ func (req *ExecuteRequest) Context(ctx context.Context) *ExecuteRequest {
 	req.ctx = ctx
 	return req
 }
+
+// protocolInfoRequest informs the server about supported protocol
+// version and features.
+type protocolInfoRequest struct {
```

Comment:
1. I don't see anything wrong with making this request public.
2. We need tests for the request, see request_test.go

Reply:
Added tests and publicity. Should we also test sending this request in runtume? (As a plain request that doesn't update any internals)

Reply:
Sure.

Reply:
Added request and request ctx test